inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/range/range_kernel_ref.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "range_kernel_ref.h"
+
+#include <kernel_selector_utils.h>
+
+namespace kernel_selector {
+
+namespace {
+
+CommonDispatchData MakeDispatchData(const range_params &params) {
+    CommonDispatchData dispatchData;
+
+    dispatchData.gws = { // TODO: these could be split better
+        params.output.Batch().v, params.output.Feature().v, params.output.Z().v * params.output.Y().v
```

Comment:
Your current realization of this kernel doesn't support "Z" dimension, so I suppose it can be deleted here.

Reply:
i've dropped everything but **x**, it exposed bug in passing dimensions in my unit tests, which i've also fixed